### ISSUE TITLE
[FR] Deletion of an experiment to mark runs associated with that experiment to be deleted

### ISSUE THREAD

**Issue by ghost** [mlflow/mlflow#5819]
### Willingness to contribute

Yes. I would be willing to contribute this feature with guidance from the MLflow community.

### Proposal Summary

Proposal - 
1. Deleting an experiment from UI to also mark runs associated with that experiment to be deleted
2. Restoring an experiment should also mark the runs to a non-deleted state.

#### What is the use case for this feature?

When I delete an experiment, the runs associated with that experiment are not marked for "delete". Hence, the artifacts associated with those runs (and indirectly the experiment) cannot be deleted (let's say, when I run mlflow garbage collection).  

#### Why is this use case valuable to support for MLflow users in general?

I want to delete those artifacts, as I don't want them in a repository permanently. I also don’t want to delete them manually.

### What component(s) does this bug affect?

- [X] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by dbczumar** [comment]
@aditya-iyengar-rtl-de Thank you for raising this feature request! We'd love to review a PR that implements this capability for the `FileStore` and the `SqlAlchemyStore`. Please let me know if you have any questions!

**Comment by aditya-iyengar-rtl-de** [comment]
@dbczumar : This feature already seems to be implemented for **"Filestore"**. 

I tested the file store by running a mlflow server locally while passing /mlruns as a value to `--backend-store-uri` i.e `mlflow ui --backend-store-uri <path_to_folder>/mlruns`). I created experiments and deleted few experiments. These deleted experiments will be stored in `/mlruns/.trash` folder.

I restored these by CLI command `$ mlflow experiments restore -x <experiment_id>` as well with `restore_experiment` API of the client module (`client.restore_experiment(experiment_id)`). 

I have implemented this feature for `SqlAlchemyStore`. I will create a Pull request and inform you about it.
